compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -1007,27 +1007,26 @@ fn place_as_reborrow(
     body: &Body<'tcx>,
     place: Place<'tcx>,
 ) -> Option<&'a [PlaceElem<'tcx>]> {
```

Comment:
This looks to me like the return type should be changed to `Option<PlaceRef>`... but that's a larger change, no need to do it in this PR.